### PR TITLE
Added an option to force a newline after an Objective C ``@end directive``  (tests should pass now)

### PR DESCRIPTION
**Pull request by Gary-Ash**
Add the option nl_oc_force_after_end to force a newline after an `@end` directive

### CODE REVIEW

**Comment by CDanU** [comment]
Pleas add tests for your new option, set that default to false and think about if it would make more sense to make that an IARF instead of a bool option.

**Comment by Gary-Ash** [comment]
I really think that this should be an a boolean option because it will either force a newline after a @end directive or not. I set the option on (force the newline by default) because I've never seen an Objective C style that doesn't have @end on its own line. I considered submitting the code

**Comment by mwoehlke-kitware** [comment]
> I really think that this should be an a boolean option because it will either force a newline after a `@end` directive or not.

That's one possible implementation. That basically makes the only options `add` or `ignore`. (It's also a possible compatibility issue if we ever change it, because an explicit `false` would change meaning from `ignore` to `remove`.) The other implementation, and the one I think @CDanU is encouraging (which I will encourage also) is to have three options:
- `add`: force a newline
- `ignore`: leave it alone
- `remove`: *remove* a newline if present

(Note: as an IARF option, `force` and `add` would mean the same. There are already some options like that.)

> I set the option on (force the newline by default) because I've never seen an Objective C style that doesn't have `@end` on its own line.

Uncrustify's general policy is to not change your code except as you ask. IOW, we prefer to default options to "no change".

(BTW, please use ``s around `@end`; otherwise it registers as a github user's name, which spams that user with irrelevant e-mail, and causes it to be capitalized oddly.)

**Comment by Gary-Ash** [comment]
changed the option to IARF

File: tests/input/oc/EndDirective.m [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+   return numerator;
+}
+@end
+@implementation SimpleTest
+-(void) print
+{
+   printf("Hello");
+}
+@end
```
Comment (mihaipopescu) [inline tests/input/oc/EndDirective.m]: I would add a final newline here.

File: tests/expected/oc/50900-EndDirective.m [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+-(int) denominator
+{
+	return denominator;
+}
+
+-(int) numerator
+{
+	return numerator;
+}
+@end
```
Comment (mihaipopescu) [inline tests/expected/oc/50900-EndDirective.m]: I don't really see how this changes from the input.

**Pull request closed without merging**